src/lib.rs

```diff
@@ -265,6 +263,7 @@ pub type Result<T, E = error::Error> = std::result::Result<T, E>;
 
 const GITHUB_BASE_URI: &str = "https://api.github.com";
 
+#[cfg(feature = "default-client")]
```

Comment:
It depends on `OctocrabBuilder::default().build()`, which is not available on wasm32. Another approach would be having 2 different `OctocrabBuilder::default()` impls.

I also found another issue with building for wasm: `Buffer::new()` [here](https://github.com/XAMPPRocky/octocrab/blob/2b68d03ceb40962e6ada8eb8412c594fb578cb78/src/lib.rs#L944) depends on tokio. To make it work on wasm I had to changed to `Buffer::pair`, pass in the executor and spawn the worker task.

Reply:
Yeah, I think I'd prefer having default client that works on WASM.

Reply:
I re-checked the code and realized why I didn't go with 2 default impls in the first place: there's currently no good default `Svc` on wasm. An empty/do-nothing one would be pointless here, since the `STATIC_INSTANCE` would not work anyway. Bringing in a working `Svc` for wasm looks like the better option, and `reqwest` is a great candidate. However it's still using an old version of `http` https://github.com/seanmonstar/reqwest/pull/2059

I made a half-working implementation using `gloo_net`, however it involves manually construct `gloo_net::Request` from `http::Request<String>`, making the code ugly. Soon after testing a few endpoint, I realize the github rest API would not work for my use case, so had to switch to its graphql API, so I didn't finish the wasm http client `tower::Service` implementation.